File: javascript/004/main.js
/*
 * ProjectEuler 004
 * JavaScript / main.js
 * copyright (c) 2015 Susisu
 *
 * Node.js v0.12.0
 * `node main.js`
 */

function main() {
    var max = 0;
    for (var x = 100; x <= 999; x++) {
        for (var y = x; y <= 999; y++) {
            var p = x * y;
            if (isPalindromic(p) && p > max) {
                max = p;
            }
        }
    }
    console.log(max);
}

function isPalindromic(n) {
    var s = n.toString();
    return s === s.split("").reverse().join("");
}

main();

File: lua/004/main.lua
--[[
    ProjectEuler 004
    Lua / main.lua
    copyright (c) 2015 Susisu

    Lua 5.1.5
    `lua main.lua`
]]

function main()
    local max = 0
    for a = 100, 999 do
        for b = 100, 999 do
            local c = a * b
            if ispalindrome(tostring(c)) and c > max then
                max = c
            end
        end
    end
    print(max)
end

function ispalindrome(str)
    return str == string.reverse(str)
end

main()
